framework/src/systems/SystemBase.C

```diff
@@ -1222,11 +1222,27 @@ SystemBase::copyVars(ExodusII_IO & io)
 
     if (hasVariable(vci._dest_name))
     {
-      if (getVariable(0, vci._dest_name).isNodal())
-        io.copy_nodal_solution(system(), vci._dest_name, vci._source_name, timestep);
-
+      if (getVariable(0, vci._dest_name).count() > 1)
+      {
+        ArrayMooseVariable & var = getFieldVariable<RealEigenVector>(0, vci._dest_name);
+        if (getVariable(0, vci._dest_name).isNodal())
+        {
+          for(unsigned int i = 0; i < getVariable(0, vci._dest_name).count(); i++)
+            io.copy_nodal_solution(system(), var.componentName(i), var.componentName(i), timestep);
```

Comment:
You can call `SubProblem::arrayVariableComponent(vci._source_name, i)` here for the source variable names in exodus file.

Reply:
@YaqiWang I have used your suggestion in my latest commit.